base_layer/wallet/src/transaction_service/protocols/transaction_send_protocol.rs

```diff
@@ -353,19 +353,21 @@ where TBackend: TransactionBackend + Clone + 'static
                 },
                 Some(send_states) => {
                     if send_states.len() == 1 {
+                        let msg_tag = send_states[0].tag;
```

Comment:
It is not needed to make a whole variable for this, it makes the code less neat

Reply:
As discussed, the `match send_states.wait_single().await {` line just after prohibits borrowing.